internal/logrus.go

```diff
@@ -48,32 +54,54 @@ func NewLogrusLogger(level string, writer io.Writer) logrusLogger {
 	return instance
 }
 
-func (logger logrusLogger) ErrorWithFields(message string, fields map[string]interface{}) {
-	entry := logger.entry(fields, stackTraceLevels[logrus.ErrorLevel])
-	entry.Error(message)
+func (logger logrusLogger) ErrorWithFields(err error, fields map[string]interface{}) {
+	var stackTrace []string
+
+	if stackTraceLevels[logrus.ErrorLevel] {
+		if err, ok := err.(stackTracer); ok {
```

Comment:
imo, this would be cleaner with a type switch
```golang
switch err := err.(type) {
case stackTracer:
   for _, f := range err.StackTrace() {
      stackTrace = append(stackTrace, fmt.Sprintf(stackTraceErrorPkgFormat, f))
   }
default:
   stackTrace = buildStackTrace()
}
```